### PR TITLE
#957 no empty poll result values

### LINKED ISSUE #957: In poll results, `0` values are always shown
They should be hidden.

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #957 
For analog polls: Hid undocumented vote results in motion poll, same for the values of votescast, votesinvalid and votesvalid in the assignment poll.
In order for this to work properly the backend needs to set the votescast, votesvalid and votesinvalid values to VOTE_UNDOCUMENTED (i.e. -2) for analog polls, if no value is sent over by the client. (This has already been discussed with @jsangmeister and @r-peschke )

Also noticed that the percentages for poll results used negative vote-values (like VOTE_UNDOCUMENTED) in their calculations (which caused frankly impossible percentages like 200% to be calculated) and fixed that as well.

### CODE REVIEW

File: client/src/app/site/polls/services/poll.service.ts [hunk 1]
```diff
@@ -267,11 +267,14 @@ export class PollService {
...
-        return (option?.yes ?? 0) + (option?.no ?? 0);
+        const yes = option?.yes ?? 0;
+        const no = option?.no ?? 0;
+        return (yes >= 0 ? yes : 0) + (no >= 0 ? no : 0);
     }
 
     private sumOptionsYNA(option: OptionData): number {
-        return this.sumOptionsYN(option) + (option?.abstain ?? 0);
+        const abstain = option?.abstain ?? 0;
+        return this.sumOptionsYN(option) + (abstain >= 0 ? abstain : 0);
```
Comment (GabrielInTheWorld) [inline client/src/app/site/polls/services/poll.service.ts:277]: Why did you change it?

File: client/src/app/site/motions/modules/motion-poll/motion-poll-dialog/motion-poll-dialog.component.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { TranslateService } from '@ngx-translate/core';
 import { PollDialogData } from 'app/core/ui-services/base-poll-dialog.service';
 import { ComponentServiceCollector } from 'app/core/ui-services/component-service-collector';
 import { BaseModel } from 'app/shared/models/base/base-model';
-import { VoteValue } from 'app/shared/models/poll/poll-constants';
+import { VOTE_UNDOCUMENTED,VoteValue } from 'app/shared/models/poll/poll-constants';
```
Comment (GabrielInTheWorld) [inline client/src/app/site/motions/modules/motion-poll/motion-poll-dialog/motion-poll-dialog.component.ts:8]: Will this be used?

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]

File: client/src/app/site/polls/services/poll.service.ts (hunk 1 above)
Comment (luisa-beerboom) [reply client/src/app/site/polls/services/poll.service.ts:277]: Because I found out that if values were set to VOTE_UNDOCUMENTED (i.e. set to -2), they would be added onto the sum, even if they really shouldn't.
The result was that the sum would be (sum of documented votes - 2*number of undocumented options).
Since percentage values are calculated on these functions this would warp them (sometimes to the point where certain options would have values like 200%).
To counteract this I changed the function to not add on values that are smaller than 0

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
It is not possible to remove a vote value, e.g.
set votes valid to "100" -> save -> now try to remove the value "100" (leave field empty) and save
=> "100" still visible. expected behavior:  empty value (= -2) should be saved.

**Comment by luisa-beerboom** [comment]
> It is not possible to remove a vote value, e.g. set votes valid to "100" -> save -> now try to remove the value "100" (leave field empty) and save => "100" still visible. expected behavior: empty value (= -2) should be saved.

This is currently being fixed in the backend ( see comments for issue #777 )

**Comment by jsangmeister** [comment]
Done in https://github.com/OpenSlides/openslides-backend/pull/1322

**Comment by luisa-beerboom** [comment]
Status report: Currently works for YNA-polls, for other polls there is a bug in the backend that currently makes it impossible to update an analog poll

**Review by GabrielInTheWorld — APPROVED** [review]

**Comment by luisa-beerboom** [comment]
> Status report: Currently works for YNA-polls, for other polls there is a bug in the backend that currently makes it impossible to update an analog poll

Working again thanks to OpenSlides/openslides-backend#1325

**Review by emanuelschuetze — APPROVED** [review]
